src/app/shared/services/json-utils.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { SchemaKeysStoreService } from './schema-keys-store.service';
+import * as _ from 'lodash';
+
+@Injectable()
+export class JsonUtilsService {
+
+  constructor(public schemaKeysStoreService: SchemaKeysStoreService) { }
+
+  filterJsonArray(jsonArray: Array<object>, filterExpression: string) {
+    let _jsonArray = [];
+    jsonArray.forEach(record => {
+      // cast to array for now before adding multiple filter expressions
+      let res = this.filterJson(record, [filterExpression]);
+      if (!_.isEmpty(res)) {
+        _jsonArray.push(res);
+      }
+    });
+    return _jsonArray;
+  }
+
+  filterJson(json: object, paths: Array<string>): object {
+    let result = {};
+    paths.forEach(path => {
+      let res = this.filterJsonRecursively(Object.assign({}, json), path);
+      if (res) {
+        _.merge(result, res);
+      }
+    });
+    this.cleanup(result);
+    return result;
+  }
+
+  private filterJsonRecursively(json: object, path: string, pathIdx = 0) {
```

Comment:
why don't you pass every time a relative path instead of passing whole path with path index?

Reply:
because all array elements have the same relative path. Is that what you mean?